### PR TITLE
Added 2FA support

### PR DESCRIPTION
**Pull request by DarkFighterLuke**
I have added 2FA authentication but introducing a specific function, so that it will be neater than using the same function adding parameters etc.

Yes, there is a little of code duplication though. Feel free to edit and tell me what to improve.

### CODE REVIEW

File: mega.go [hunk 1]
```diff
@@ -531,6 +531,9 @@ func (m *Mega) login(email string, passwd string) error {
 
 	msg[0].Cmd = "us"
 	msg[0].User = email
+	if multiFactor != "" {
```
Comment (ncw) [inline mega.go:534]: The if statement isn't necessary here since the default for `msg[0].Mfa` will be `""`

**Review by ncw — COMMENTED** [review]
Sorry for the delay in reviewing.

Can you remove the code duplication? Make the original login be `MultiFactorLogin` with the extra parameter, then make a `Login` with the same parameters as the existing `Login` which just calls `MultiFactor` login with the extra parameter as "".

**Comment by DarkFighterLuke** [comment]
I've just removed code duplication as per your request.

Tell me if it's good.

**Comment by AtilioA** [comment]
@ncw is this good? Don't want this to stale into oblivion :)

**Review by janreggie — APPROVED** [review]

**Comment by janreggie** [comment]
I don't think this repo is dead. A change was made just 17 days ago.

bump.

**Comment by DarkFighterLuke** [comment]
> I don't think this repo is dead. A change was made just 17 days ago.
> 
> bump.

Yes, I noticed that, but for sure this project is not actively mantained, it is more dead than alive as you can see from PR management.

This PR was made an year ago, some changes were requested and promptly applied, but it is staled at the moment...

**Comment by lewer** [comment]
Any chance this PR gets merged? It's needed for `rclone`, see https://github.com/rclone/rclone/issues/3165. Thanks.

**Comment by ncw** [comment]
Sorry for the delay with this :-(

I'll merge this request now as I think it looks great!

**Pull request merged**